File: hi.py
class VulnerableFunction:

    def main(self):

        source = "too long"
        dest = [""] * 5
        trigger = 5

        if trigger == 5:

            # Simulated unsafe copy
            for i in range(len(source)):
                dest[i] = source[i]

        return 0


if __name__ == "__main__":
    vf = VulnerableFunction()
    vf.main()

File: hi.java
public class BufferOverflowDemo {
    public static void main(String[] args) {

        String source = "too long";
        char[] dest = new char[5];
        int trigger = 5;

        if (trigger == 5) {

            // Simulating unsafe copy
            for (int i = 0; i < source.length(); i++) {
                dest[i] = source.charAt(i);
            }
        }

        System.out.println("Done");
    }
}
